File: core/src/main/java/me/liam/support/SupportFragmentVisible.java
package me.liam.support;

import android.os.Bundle;

import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

class SupportFragmentVisible {

    final public static String FRAGMENTATION_IS_HIDDEN = "Fragmentation:IsHidden";

    private SupportFragment supportFragment;

    public void onAttach(SupportFragment supportFragment) {
        this.supportFragment = supportFragment;
    }

    public void onActivityCreated(Bundle savedInstanceState) {
        if (savedInstanceState != null){
            if (savedInstanceState.getBoolean(FRAGMENTATION_IS_HIDDEN)){
                FragmentManager fm = supportFragment.getFragmentManager();
                FragmentTransaction ft = fm.beginTransaction();
                ft.hide(supportFragment);
                ft.commitAllowingStateLoss();
            }
        }
    }

    public void onSaveInstanceState(Bundle outState) {
        outState.putBoolean(FRAGMENTATION_IS_HIDDEN,supportFragment.isHidden());
    }
}

File: core/src/main/java/me/liam/fragmentation/support/SupportFragmentVisible.kt
package me.liam.fragmentation.support

import android.os.Bundle

/**
 * 管理Fragment的显示和隐藏状态
 */
internal class SupportFragmentVisible {
    private var supportFragment: SupportFragment? = null

    fun onAttach(supportFragment: SupportFragment) {
        this.supportFragment = supportFragment
    }

    fun onActivityCreated(savedInstanceState: Bundle?) {
        if (savedInstanceState?.getBoolean(FRAGMENTATION_IS_HIDDEN) == true) {
            supportFragment?.fragmentManager?.beginTransaction()?.apply {
                supportFragment?.let { hide(it) }
                commitAllowingStateLoss()
            }
        }
    }

    fun onSaveInstanceState(outState: Bundle) {
        outState.putBoolean(FRAGMENTATION_IS_HIDDEN, supportFragment?.isHidden ?: false)
    }

    companion object {
        const val FRAGMENTATION_IS_HIDDEN = "Fragmentation:IsHidden"
    }
}
